### DISCUSSION TITLE
Logging custom values

### DISCUSSION THREAD

**Discussion by varun-nambiar** [adonisjs/core#1505] [Help] [answered]
If I need to log a JSON/Object data do I need to do the following?
```
const user = await User.find(1);
logger.info(`user: ${JSON.stringify(user)}`)
```

Doing the following doesn't log the 'user' object
```
logger.info('data', user)
```

Do I need to add any config to the logger in config/app

I have been using console.log but I really like the logger contract for its timestamp and so planning to redo all those logs into logger, But I still need to log in objects and not just strings

Thank you for reading!

**Comment by thetutlage** [comment, marked as answer]
Which version of the framework are you using?

**Reply by varun-nambiar** [reply]
> Which version of the framework are you using?

The preview version - v5
`"@adonisjs/core": "^5.0.0-preview-rc-1.9",`

**Reply by thetutlage** [reply]
When logging, you need to use the sprintf placeholders for values to appear.

```ts
Logger.info('user %o', user)
```

Or you can pass values as a [merging object](https://getpino.io/#/docs/api?id=mergingobject-object). Following is an example of the merging object 

```ts
Logger.info({ user }, 'user %o')
```

**Reply by varun-nambiar** [reply]
Thank you!
